### PR TITLE
Update npm

### PR DESCRIPTION
**Pull request by adriangabura**

### CODE REVIEW

**Comment by adriangabura** [comment]
@au-re So I updated the package.json by using npm i package for each package. We would need to decide what to do about major version changes for things like react redux, webpack

File: package.json [hunk 1]
```diff
@@ -116,13 +116,13 @@
   },
   "dependencies": {
     "@simonwep/selection-js": "^1.6.0",
```
Comment (adriangabura) [inline package.json:118]: @au-re This package is deprecated. The author has changed the name. https://github.com/Simonwep/selection 
2.x doesn't work with Ramen.

File: package.json [hunk 2]
```diff
@@ -53,60 +53,60 @@
...
+    "storybook-readme": "^5.0.9",
+    "styled-components": "^5.3.6",
+    "terser": "^5.16.1",
+    "ts-jest": "^25.5.1",
     "tslint": "^5.20.1",
     "tslint-config-airbnb": "^5.11.2",
     "tslint-react": "^4.1.0",
-    "typescript": "^3.7.2",
-    "webpack-cli": "^3.3.10"
+    "typescript": "^3.9.10",
```
Comment (adriangabura) [inline package.json:108]: Typescript can't be upgraded because awesome-typescript-loader depends on typescript 2 and 3.

File: package.json [hunk 3]
```diff
@@ -53,60 +53,60 @@
     }
   },
   "peerDependencies": {
-    "react": "^16.0.0",
-    "react-dom": "^16.0.0",
-    "styled-components": "^5.0.0"
+    "react": "^16.14.0",
```
Comment (adriangabura) [inline package.json:56]: react-split-pane prevents the upgrade of react to a higher version

File: package.json [hunk 4]
```diff
@@ -53,60 +53,60 @@
...
-    "@types/react-test-renderer": "^16.9.1",
-    "@types/simonwep__selection-js": "^1.2.0",
-    "@types/storybook-readme": "5.0.4",
-    "@types/styled-components": "5.0.1",
-    "ace-builds": "^1.4.7",
+    "@babel/core": "^7.20.2",
+    "@babel/plugin-transform-runtime": "^7.19.6",
+    "@storybook/addon-options": "^5.3.21",
+    "@storybook/addons": "^5.3.22",
+    "@storybook/react": "^5.3.21",
```
Comment (adriangabura) [inline package.json:65]: When attempting to upgrade to 6th version the following error appears:
Error: You have both a "main" and a "config". Please remove the "config" file from your configDir (/mnt/fdf8f571-43ba-4785-b8dd-16c63e8595f9/projects/ramen/.storybook/config)

File: package.json [hunk 5]
```diff
@@ -53,60 +53,60 @@
...
+    "@storybook/theming": "^6.5.14",
+    "@types/jest": "^29.2.4",
+    "@types/lodash.get": "^4.4.7",
+    "@types/lodash.isequal": "^4.5.6",
+    "@types/react": "^16.14.34",
```
Comment (adriangabura) [inline package.json:70]: This depends on upgrading React

File: package.json [hunk 6]
```diff
@@ -53,60 +53,60 @@
...
+    "@types/styled-components": "^5.1.26",
+    "ace-builds": "^1.13.2",
     "awesome-typescript-loader": "^5.2.1",
-    "babel-loader": "^8.0.6",
-    "browserify": "^16.5.0",
-    "commitizen": "^4.0.4",
-    "cz-emoji": "^1.1.2",
-    "gh-pages": "^2.2.0",
-    "jest": "^25.1.0",
+    "babel-loader": "^8.3.0",
```
Comment (adriangabura) [inline package.json:78]: Cannot be upgraded because @storybook/react depends on this version

File: package.json [hunk 7]
```diff
@@ -53,60 +53,60 @@
...
-    "terser": "^4.3.9",
-    "ts-jest": "^25.2.1",
+    "rollup-plugin-terser": "^7.0.2",
+    "rollup-plugin-typescript2": "^0.34.1",
+    "source-map-explorer": "^2.5.3",
+    "source-map-loader": "^1.0.0",
+    "storybook-readme": "^5.0.9",
+    "styled-components": "^5.3.6",
+    "terser": "^5.16.1",
+    "ts-jest": "^25.5.1",
```
Comment (adriangabura) [inline package.json:104]: Can't upgrade ts-jest because of typescript

File: package.json [hunk 8]
```diff
@@ -53,60 +53,60 @@
...
-    "gh-pages": "^2.2.0",
-    "jest": "^25.1.0",
+    "babel-loader": "^8.3.0",
+    "browserify": "^17.0.0",
+    "commitizen": "^4.2.6",
+    "cz-emoji": "^1.3.1",
+    "gh-pages": "^4.0.0",
+    "jest": "^25.5.4",
     "lodash.noop": "^3.0.1",
     "markdown-loader": "^5.1.0",
```
Comment (adriangabura) [inline package.json]: (Use `node --trace-deprecation ...` to show where the warning was created)
/mnt/fdf8f571-43ba-4785-b8dd-16c63e8595f9/projects/ramen/node_modules/html-webpack-plugin/lib/webpack5/file-watcher-api.js:13
    mainCompilation.fileSystemInfo.createSnapshot(
                                   ^

TypeError: Cannot read properties of undefined (reading 'createSnapshot')
    at /mnt/fdf8f571-43ba-4785-b8dd-16c63e8595f9/projects/ramen/node_modules/html-webpack-plugin/lib/webpack5/file-watcher-api.js:13:36
    at new Promise (<anonymous>)
    at Object.createSnapshot (/mnt/fdf8f571-43ba-4785-b8dd-16c63e8595f9/projects/ramen/node_modules/html-webpack-plugin/lib/webpack5/file-watcher-api.js:12:10)
    at /mnt/fdf8f571-43ba-4785-b8dd-16c63e8595f9/projects/ramen/node_modules/html-webpack-plugin/lib/cached-child-compiler.js:219:35

File: package.json [hunk 9]
```diff
@@ -53,60 +53,60 @@
...
-    "source-map-explorer": "^2.1.0",
-    "source-map-loader": "^0.2.4",
-    "storybook-readme": "^5.0.8",
-    "styled-components": "^5.1.0",
-    "terser": "^4.3.9",
-    "ts-jest": "^25.2.1",
+    "rollup-plugin-terser": "^7.0.2",
+    "rollup-plugin-typescript2": "^0.34.1",
+    "source-map-explorer": "^2.5.3",
+    "source-map-loader": "^1.0.0",
```
Comment (adriangabura) [inline package.json:100]: n is deprecated. Use new API on `.hooks` instead
(Use `node --trace-deprecation ...` to show where the warning was created)
/mnt/fdf8f571-43ba-4785-b8dd-16c63e8595f9/projects/ramen/node_modules/html-webpack-plugin/lib/webpack5/file-watcher-api.js:13
    mainCompilation.fileSystemInfo.createSnapshot(
                                   ^

TypeError: Cannot read properties of undefined (reading 'createSnapshot')
    at /mnt/fdf8f571-43ba-4785-b8dd-16c63e8595f9/projects/ramen/node_modules/html-webpack-plugin/lib/webpack5/file-watcher-api.js:13:36
    at new Promise (<anonymous>)
    at Object.createSnapshot (/mnt/fdf8f571-43ba-4785-b8dd-16c63e8595f9/projects/ramen/node_modules/html-webpack-plugin/lib/webpack5/file-watcher-api.js:12:10)
    at /mnt/fdf8f571-43ba-4785-b8dd-16c63e8595f9/projects/ramen/node_modules/html-webpack-plugin/lib/cached-child-compiler.js:219:35

File: package.json [hunk 10]
```diff
@@ -53,60 +53,60 @@
...
+    "storybook-readme": "^5.0.9",
+    "styled-components": "^5.3.6",
+    "terser": "^5.16.1",
+    "ts-jest": "^25.5.1",
     "tslint": "^5.20.1",
```
Comment (adriangabura) [inline package.json:105]: Can't upgrade because of   "tslint-config-airbnb": "^5.11.2",

**Review by au-re — APPROVED** [review]
thanks for taking the time to do this upgrade! I will look at how to get the rest upgraded as well

**Pull request merged**